src/DFG.cpp

```diff
@@ -49,6 +38,75 @@ DFG::DFG(Function& t_F, list<Loop*>* t_loops, bool t_targetFunction,
 
 }
 
+// Specilized fusion for the nonlinear operations.
+void DFG::nonlinear_combine() {
+  tuneForBitcast();
+  combineMulAdd(2);
+  combinePhiAdd(1);
+  combine("fcmp", "select", 1);
+  combine("icmp", "select", 1);
+  combine("icmp", "br", 2);
+  combine("fcmp", "br", 2);
+  combineAddAdd(1);
+  tuneForPattern();
+  tuneDivPattern();
+}
+
+// For division, we regrad it as non-vectorized instructions, which is contradictory to LLVM Pass.
+// Thus we need to split a vectorization divison into multiple scalar divisions.
+void DFG::tuneDivPattern() {
+  list<DFGNode*>* removeNodes = new list<DFGNode*>();
+  list<DFGNode*>* splitNodes = new list<DFGNode*>();
+  int dfgNodeID = nodes.size();
+  for (DFGNode* dfgNode: nodes) {
+    cout << "tuneDivPattern:" << dfgNode->isDiv() << " " << dfgNode->isVectorized() << "\n";
+    if (dfgNode->isOpt("sdiv") && dfgNode->isVectorized()) {
```

Comment:
Are float div and int div same in LLVM IR? We only care `sdiv` here?

Reply:
Not the same. `sdiv` is for integer and in my design we only enable vectorization of integer division, thus I only care `sdiv` here.

Reply:
Not the same. `sdiv` is for integer and in my design we only enable vectorization of integer, thus we only need to split integer vectorization into multiple nodes. For floating-point arithmetic, there is no vectorization thus we dont need it.